### PR TITLE
Sensor Things API init API

### PR DESCRIPTION
**Pull request by KVISDAOWNER**
# Related Issue
First merge of backend into develop. First successfull build.

# Pull Request Checklist
- [ ] Tests
- [x] Documentation

# For code reviewers
Checks to make during review of pull request.
## General
- [ ] Does the code work? Does it perform its intended function, the logic is correct etc.
- [ ] Is all the code easily understood?
- [ ] Is the code well commented?
- [ ] Is there any redundant or duplicate code?
- [ ] Are good names used for classes, enum, struct , methods, and variables??
- [ ] Can any global variables be replaced?
- [ ] Can any logging or debugging code be removed?
- [ ] Can any of the code be replaced with library functions?
- [ ] Does code follow defined architecture?
- [ ] Does
## Testing
- [ ] Is the code testable? i.e. don’t add too many or hide dependencies, unable to initialize objects, test frameworks can use methods etc.
- [ ] Do tests exist and are they comprehensive? i.e. has at least your agreed on code coverage.
- [ ] Do unit tests actually test that the code is performing the intended functionality?
- [ ] Are arrays checked for ‘out-of-bound’ errors?
- [ ] Could any test code be replaced with the use of an existing API?

### CODE REVIEW

**Comment by KVISDAOWNER** [comment]
There is a lack of unit tests in this branch, but this intial build is functional and partly tested through postman. #3 has been created to address this.

File: backend/SensorThingsAPI/src/Org.OpenAPITools/Attributes/ValidateModelStateAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /// </summary>
+    public class ValidateModelStateAttribute : ActionFilterAttribute
+    {
+        /// <summary>
+        /// Called before the action method is invoked
+        /// </summary>
+        /// <param name="context"></param>
+        public override void OnActionExecuting(ActionExecutingContext context)
+        {
+            // Per https://blog.markvincze.com/how-to-validate-action-parameters-with-dataannotation-attributes/
```
Comment (Bjerkk) [inline backend/SensorThingsAPI/src/Org.OpenAPITools/Attributes/ValidateModelStateAttribute.cs]: Hvem er Per?

File: backend/SensorThingsAPI/src/Org.OpenAPITools/Authentication/ApiAuthentication.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc.Filters;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Org.OpenAPITools.Authentication
+{
+    public class ApiKeyRequirement : IAuthorizationRequirement
```
Comment (Bjerkk) [inline backend/SensorThingsAPI/src/Org.OpenAPITools/Authentication/ApiAuthentication.cs]: Skal vi bruge den her class?

File: backend/SensorThingsAPI/src/Org.OpenAPITools/Attributes/ValidateModelStateAttribute.cs (hunk 1 above)
Comment (KVISDAOWNER) [reply backend/SensorThingsAPI/src/Org.OpenAPITools/Attributes/ValidateModelStateAttribute.cs]: Good catch. This is auto generated by openAPI so lets just leave it for potential future use.

File: backend/SensorThingsAPI/src/Org.OpenAPITools/Authentication/ApiAuthentication.cs (hunk 2 above)
Comment (KVISDAOWNER) [reply backend/SensorThingsAPI/src/Org.OpenAPITools/Authentication/ApiAuthentication.cs]: No, this should be deleted.

**Comment by KVISDAOWNER** [comment]
> Navnet på projektet skal ændres

This has been done

**Comment by KVISDAOWNER** [comment]
Initial SensorThingsAPI merge

**Comment by KVISDAOWNER** [comment]
Error fixed for merge base, it should be develop

**Pull request merged**